src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
+  createInput('salary', 'number'),
+  createButton('Save to table', 'submit'),
+];
+
+// CREATE FORM AND ADD ELEMENTS
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+formElements.forEach((element) => form.append(element));
+document.addEventListener('DOMContentLoaded', () => document.body.append(form));
+
+// FORM VALIDATION
+const positionValidatorFunction = (value) =>
+  !value
+    ? { validated: false, message: `Position is a required field!` }
+    : { validated: true };
+
+const nameValidatorFunction = (value) =>
+  !value || value.length < 4
+    ? { validated: false, message: 'Name should be more than 4 chars long' }
+    : { validated: true };
+
+const ageValidatorFunction = (value) =>
+  value < 18 || value > 90
+    ? { validated: false, message: 'Age must be between 18 and 90' }
+    : { validated: true };
+
+const getValidatorFunction = (value, key) => {
+  const validators = {
+    position: positionValidatorFunction,
+    name: nameValidatorFunction,
+    age: ageValidatorFunction,
+  };
+
+  return validators[key]?.(value) || { validated: true };
+};
+
+const validateForm = (data) => {
+  return Array.from(data)
+    .map(([key, value]) => getValidatorFunction(value, key))
+    .filter((result) => !result.validated)
+    .map((result) => result.message);
+};
+
+// NOTIFICATIONS ON SUCESS / ERROR
+const displaySuccessNotice = (message) => {
+  displayNotification('success', 'Success!', [message]);
+};
+
+const displayErrorNotice = (errorArray) => {
+  displayNotification('error', 'Error!', errorArray);
+};
+
+const displayNotification = (type, titleText, errorMessageArr) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.innerHTML = `
+    <h3 class="title">${titleText}</h3>
+    <p>
+      <ul>
+        ${errorMessageArr.map((message) => `<li>${message}</li>`).join('')}
+      </ul>
+    </p>
+  `;
+
+  document.body.append(notification);
+  setTimeout(() => notification.remove(), 2000);
+};
+
+// HANDLE FORM SUBMIT EVENT
+const appendTableWithData = (data) => {
+  const newRow = tBody.insertRow(-1);
+
+  data.forEach((value, key) => {
+    newRow.insertCell(-1).textContent =
+      key === 'salary' ? window.convertNumberToSalary(value) : value;
+  });
+};
+
+const handleSubmitFormEvent = (e) => {
+  e.preventDefault();
+
+  const target = e.target.closest('form');
+  const formData = new FormData(target);
+  const errors = validateForm(formData);
+
+  if (errors.length === 0) {
+    const message = 'New employee has been added to the list!';
+
+    appendTableWithData(formData);
+    displaySuccessNotice(message);
+  } else {
+    displayErrorNotice(errors);
+  }
+};
+
+form.addEventListener('submit', handleSubmitFormEvent);
+// #endregion
+
+// #region CELL_EDITTING
+// EVENT HANDLERS TO SAVE EDITED DATA
+const saveCellInput = (cell, value) => {
```

Comment:
The `saveCellInput` function checks if the cell is an instance of `HTMLTableCellElement`. This check is unnecessary because the function is only ever called with a `td` element. You can remove this check to simplify the code.